src/components/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,116 @@
+
+export const AppConfig = () => {
+  const s = useStyles2(getStyles);
+  const { checkTypes, isLoading, isError } = useCheckTypes();
+  const { updateIgnoreStepsAnnotation, updateCheckTypeState } = useSkipCheckTypeStep();
+
+  return (
+    <FieldSet label="Available Check Types">
+      <div className={s.description}>Enable or disable the steps for each check type.</div>
+      {isLoading && <LoadingPlaceholder text="Loading check types..." />}
+      {isError && <div>Error loading check types</div>}
+      {!isLoading && !isError && checkTypes && checkTypes.length === 0 && <div>No check types available</div>}
+      {!isLoading && !isError && checkTypes && checkTypes.length > 0 && (
+        <div className={s.checkTypesList}>
+          {checkTypes.map((checkType) => {
+            const typeName = checkType.metadata.name!;
+            const canIgnoreSteps = checkType.metadata.annotations?.[IGNORE_STEPS_ANNOTATION] !== '';
+            const ignoreSteps =
+              checkType.metadata.annotations?.[IGNORE_STEPS_ANNOTATION_LIST]?.split(',').filter(Boolean) || [];
+            return (
+              <Card key={typeName} className={s.checkTypeCard}>
+                <Card.Heading>Check type: {checkType.spec.name}</Card.Heading>
+                <Card.Description>
+                  <div>Steps:</div>
+                  <ul className={s.stepsList}>
+                    {checkType.spec.steps.map((step) => (
+                      <li key={step.stepID} className={s.stepItem}>
+                        <Stack direction="row">
+                          <div className={s.switchWrapper}>
+                            <Switch
+                              value={!ignoreSteps.includes(step.stepID)}
+                              onChange={(e) => {
+                                const ignore = !e.currentTarget.checked;
+                                if (ignore) {
+                                  ignoreSteps.push(step.stepID);
+                                } else {
+                                  ignoreSteps.splice(ignoreSteps.indexOf(step.stepID), 1);
+                                }
+                                updateIgnoreStepsAnnotation(typeName, ignoreSteps);
+                              }}
+                              disabled={!canIgnoreSteps || updateCheckTypeState.isLoading}
+                            />
+                          </div>
+                          <div className={s.stepDescription}>
+                            <strong>{step.title}</strong> - {step.description}
+                          </div>
+                        </Stack>
+                      </li>
+                    ))}
+                  </ul>
+                  {!canIgnoreSteps && (
+                    <div className={s.missingAnnotationNote}>
+                      Your current version of Grafana does not support ignoring steps
+                    </div>
+                  )}
+                </Card.Description>
+              </Card>
```

Comment:
I think we would benefit from refactoring this into two smaller components, one for the list and one for each item